pandas/tseries/period.py

```diff
@@ -187,6 +190,16 @@ def __new__(cls, data=None, ordinal=None, freq=None, start=None, end=None,
         if name is None and hasattr(data, 'name'):
             name = data.name
 
+        if dtype is not None:
+            dtype = pandas_dtype(dtype)
```

Comment:
pickle already contains normal ``PeriodIndex``, is there any type specifically care?

- https://github.com/pydata/pandas/blob/master/pandas/io/tests/generate_legacy_storage_files.py#L75